### DISCUSSION TITLE
Issue with installing cfclient on m2 Mac

### DISCUSSION THREAD

**Discussion by hohyunLee96** [bitcraze/discussions#1442] [Q&A] [answered]
Dear developers,

I`m new to the cfclient and I was trying to install the cfclient on my Mac and I followed the instructions. 

I have run this "python3 -m pip install cfclient" on my terminal and then it said that I can run the client with "cfclient" or "python3 -m cfclient.gui".

But every time I try to launch the client with any of the commands above, the error displays. 
and the error is 

> UserWarning: Using SDL2 binaries from pysdl2-dll 2.24.0
INFO:cfclient.gui:Disabling STL printouts
INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
INFO:cfclient.utils.input.inputreaders:Could not initialize [linuxjsdev]: Only supported on Linux
INFO:cfclient.utils.input.inputreaders:Successfully initialized [pysdl2]
INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
INFO:cfclient.utils.config:Dist config read from /opt/anaconda3/lib/python3.11/site-packages/cfclient/configs/config.json
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file
INFO:OpenGL.acceleratesupport:No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate'
INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file
INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file
INFO:cfclient.gui:If startup fails because of "xcb", install dependency with `sudo apt install libxcb-xinerama0`.
qt.qpa.plugin: Could not load the Qt platform plugin "cocoa" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
Available platform plugins are: offscreen, minimal, cocoa.
zsh: abort      cfclient

do any of you know how to fix this error? I tried to solve it by installing dependencies or reinstalling the cfclient but couldn`t solve the issue.

**Comment by ataffanel** [comment]
What version of python are you using and how was it installed?

I just tried to ```pip install cfclient``` both with brew's Python 3.12.4 and MacOS Sonoma's Python 3.9.6 and I did not get this error, the client could start properly.

**Reply by hohyunLee96** [reply]
Hello, you said that you use python 3.12.4 so I updated my python version to 3.12.4. but it couldn`t solve the issue.

**Reply by ataffanel** [reply, marked as answer]
I have been running from a venv to make sure things where clean, maybe that would help?
```
$ python3 -m venv vevn
$ . ./venv/bin/activate
$ pip install cfclient
$ cfclient
```

**Reply by hohyunLee96** [reply]
Thanks, it worked :)
